File: 129. Sum Root to Leaf Numbers.java
class Solution {
    int sum;
    public int sumNumbers(TreeNode root) {
        cal(root, 0);
        return sum;
    }

    public void cal(TreeNode root, int parent ) {
        if(root == null) return;
        if(root.left == null && root.right == null)
            sum += (parent * 10) + root.val;
        else {
            cal(root.left, (parent * 10) + root.val);
            cal(root.right, (parent * 10) + root.val);
        }
    }
}

File: 129. Sum Root to Leaf Numbers.kt
class Solution {
    val ans = mutableListOf<Int>()
    fun sumNumbers(root: TreeNode?): Int {
        if(root == null) {
            return -1
        }
        subNum(root, "")
        return ans.sum()
    }
    fun subNum(root: TreeNode, sb: String){
        if(root.left == null && root.right == null)  {
            ans.add((sb + root.`val`).toInt() )
            return
        }
        if(root.left != null) {
            subNum(root.left, sb + root.`val`)
        }
        if(root.right != null) {
            subNum(root.right, sb + root.`val`)
        }
    }
}
